src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2693 @@
+
+                    // Minimum threshold for becoming a baker.
+                    Amount minimum_threshold_for_baking = 10;
+                    // Keys allowed to do root updates.
+                    HigherLevelKeys root_keys = 11;
+                    // Keys allowed to do level1 updates;
+                    HigherLevelKeys level1_keys = 12;
+                    // Keys allowed to do parameter updates.
+                    AuthorizationsV0 level2_keys = 13;
+        -}
+        return Parameters.ChainParameters{..}
+
+instance FromProto Proto.ChainParametersV1 where
+    type Output' Proto.ChainParametersV1 = Parameters.ChainParameters' 'ChainParametersV1
+    fromProto cParams = do
+        _cpElectionDifficulty <- fromProto =<< cParams ^? ProtoFields.electionDifficulty
+        _cpExchangeRates <- do
+            euroPerEnergy <- fromProto =<< cParams ^? ProtoFields.euroPerEnergy
+            microCcdPerEuro <- fromProto =<< cParams ^? ProtoFields.microCcdPerEuro
+            return $ Parameters.makeExchangeRates euroPerEnergy microCcdPerEuro
+        _cpCooldownParameters <- fromProto =<< cParams ^? ProtoFields.cooldownParameters
+        _cpTimeParameters <- Nothing -- FIXME: I do not see this (required field) anywhere in the protobuf message.
+        _cpAccountCreationLimit <- fromProto =<< cParams ^? ProtoFields.accountCreationLimit
+        _cpRewardParameters <- do
+            _rpMintDistribution <- fromProto =<< cParams ^? ProtoFields.mintDistribution
+            -- \|Distribution of transaction fees.
+            _rpTransactionFeeDistribution <- fromProto =<< cParams ^? ProtoFields.transactionFeeDistribution
+            -- \|Rewards paid from the GAS account.
+            _rpGASRewards <- fromProto =<< cParams ^? ProtoFields.gasRewards
+            return Parameters.RewardParameters{..}
+        -- FIXME: ProtoFields.foundationAccount has type AccountAddress, but the type expects an AccountIndex.
+        _cpFoundationAccount <- Nothing
+        _cpPoolParameters <- Nothing
+        return Parameters.ChainParameters{..}
+
+instance FromProto Proto.Epoch where
+    type Output' Proto.Epoch = Epoch
+    fromProto = return . deMkWord64
+
+instance FromProto Proto.CredentialsPerBlockLimit where
+    type Output' Proto.CredentialsPerBlockLimit = CredentialsPerBlockLimit
+    fromProto = deMkWord16
+
+-- FIXME: Pertains to the next FIXME comment.
+data ChainParameterOutput
+    = ChainParameterOutputV0 !(Parameters.ChainParameters' 'ChainParametersV0)
+    | ChainParameterOutputV1 !(Parameters.ChainParameters' 'ChainParametersV1)
+
+instance FromProto Proto.ChainParameters where
+    type
+        -- FIXME: Perhaps there is some nice(r) way achieve this
+        --        with (constrained) type variables or type
+        --        families? But I am uncertain whether the class
+        --        has to change, or if this can be done "locally"?
+        --        Or does it not matter?
```

Comment:
I don't quite see what you want here. You could instead return an existential type, but I don't think that is better. In fact I think it's worse.